sync/class.jetpack-sync-module-posts.php

```diff
@@ -316,76 +286,84 @@ function filter_post_content_and_add_links( $post_object ) {
 	}
 
 	public function save_published( $new_status, $old_status, $post ) {
-		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
-			$this->just_published[ $post->ID ] = true;
-		}
 
-		$this->previous_status[ $post->ID ] = $old_status;
+
+		if ( ! isset( $this->sync_items[ $post->ID ] ) ) {
+			$this->sync_items[ $post->ID ] = new Jetpack_Sync_Item( 'save_post' );
+		}
+		$sync_item = $this->sync_items[ $post->ID ];
+		$just_published = 'publish' === $new_status && 'publish' !== $old_status;
```

Comment:
Consider using variable names that start with `is_` to make it clear that they hold boolean values, e.g. `$is_just_published`